operator/controllers/keptntask/function_utils.go

```diff
@@ -20,7 +22,7 @@ type FunctionExecutionParams struct {
 
 func (r *KeptnTaskReconciler) generateFunctionJob(task *klcv1alpha1.KeptnTask, params FunctionExecutionParams) (*batchv1.Job, error) {
 	randomId := rand.Intn(99999-10000) + 10000
-	jobId := fmt.Sprintf("klc-%s-%s-%s-%d", task.Spec.AppName, task.Spec.Workload, task.Spec.WorkloadVersion, randomId)
+	jobId := fmt.Sprintf("klc-%s-%d", common.TruncateString(task.Name, 50), randomId)
```

Comment:
why 50? I'd add it as constant in common and provide a comment on the reasoning around each magic number :)